trio/_highlevel_open_tcp_stream.py

```diff
@@ -114,8 +116,12 @@ def close_all():
                 sock.close()
             except BaseException as exc:
                 errs.append(exc)
-        if errs:
-            raise trio.MultiError(errs)
+        if len(errs) == 1:
+            raise errs[0]
+        elif errs:
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
You don't need this warnings filter anymore. Also it looks like you might need a `# pragma: no cover` on this block.